tests/www/test_views.py

```diff
@@ -3293,3 +3294,25 @@ def test_action_logging_post(self):
         self.check_last_log(
             "example_bash_operator", event="clear", execution_date=self.EXAMPLE_DAG_DEFAULT_DATE
         )
+
+
+class TestHelperFunctions(unittest.TestCase):
+    @parameterized.expand(
+        [
+            ("http://google.com", "/home"),
+            (
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%test_dag';alert(33)//",
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%25test_dag",
```

Comment:
```suggestion
                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%25test_dag&alert%2833%29%2F%2F=",
```

Reply:
fixed in https://github.com/apache/airflow/pull/12459/commits/7486153f451e4d2bb1c6fd9cbb5a63430157c99c